### PR TITLE
Billing report revived

### PR DESCRIPTION
**Pull request by ehlertjd**
See original PR #1475 for details.

This is a (somewhat) breaking change as it replaces the usage endpoint.

### Review Checklist

- Tests were added to cover all code changes
- Is there a DB upgrade or fix?
  - If so is it covered by integeration tests and run against dev?
- Would this PR benefit from a test plan? If so, is one provided?
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md
- Extra care is taken and log messages are created for [Critical Data](https://github.com/flywheel-io/core/wiki/Critical-Data-Flows)

### CODE REVIEW

File: api/reports/handler.py [hunk 1]
```diff
@@ -12,37 +20,90 @@ def get_types(self):
...
+                write = start_response('200 OK', [
+                    ('Content-Type', report_writer.get_content_type()),
+                    ('Content-Disposition', 'attachment; filename="{}"'.format(report_writer.get_filename())),
+                    ('Connection', 'keep-alive')
+                ])
+
+                report_writer.execute(write)
+                return ''
+
+            return response_handler
```
Comment (davidfarkas) [inline api/reports/handler.py]: maybe we could use the `util.build_sse_handler` here

File: api/reports/handler.py [hunk 2]
```diff
@@ -12,37 +20,90 @@ def get_types(self):
...
+                        log.info('SSE upload progress failed to send; continuing')
+
+                # Log last collection time
+                time_since = datetime.datetime.now() - datetime.datetime(1970, 1, 1)
+                values.LAST_REPORT_COLLECTION.labels(report_type).set(time_since.total_seconds())
+            except Exception: # pylint: disable=broad-except
+                log.exception('Error collecting %s report data', report_type)
+                values.REPORT_COLLECTION_ERROR_COUNT.labels(report_type).inc()
+
+            return ''
```
Comment (davidfarkas) [inline api/reports/handler.py]: same here (`util.build_sse_handler`)

File: api/reports/usage_report.py [hunk 3]
```diff
@@ -67,300 +114,380 @@ def user_can_generate(self, uid, roles):
         has_privilege(roles, self.required_role)
```
Comment (davidfarkas) [inline api/reports/usage_report.py]: I know you didn't touch this, but this is not necessary anymore.

File: swagger/schemas/output/report-daily-usage.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            "group_job_count": 1,
+            "center_storage_bytes": 0,
+            "year": 2018,
+            "month": 12,
+            "group_storage_bytes": 2048,
+            "center_compute_ms": 0,
+            "day": 11
+        }
+    ]
+}
```
Comment (davidfarkas) [inline swagger/schemas/output/report-daily-usage.json]: not nitpicking, I just noticed missing new line :)

File: swagger/schemas/output/report-legacy-usage.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+            "_id": "576be50c59b2b6346719ee4d",
+            "label": "Anxiety Study"
+        },
+        "file_mbs": 3722.1714448928833,
+        "session_count": 9,
+        "gear_execution_count": 708
+    }]
+}
+
+
```
Comment (davidfarkas) [inline swagger/schemas/output/report-legacy-usage.json]: same here but with extra new line.

File: swagger/schemas/output/report-usage.json [hunk 6]
```diff
@@ -2,15 +2,38 @@
...
+            "center_job_count": 2,
+            "project": "5c102a424026b30019f9fc67",
+            "group_job_count": 2,
+            "center_storage_byte_day": 18432,
+            "session_count": 5,
+            "total_storage_byte_day": 46080,
+            "group_storage_byte_day": 27648
+        }
+    ]
+}
```
Comment (davidfarkas) [inline swagger/schemas/output/report-usage.json]: new line

File: api/reports/handler.py (hunk 1 above)
Comment (ehlertjd) [reply api/reports/handler.py]: For this one we are using different headers from SSE, but agree on the next one.

File: api/reports/usage_report.py [hunk 7]
```diff
@@ -1,366 +1,492 @@
...
+                    count_key = 'center_job_count'
+                    time_key = 'center_compute_ms'
+                else:
+                    count_key = 'group_job_count'
+                    time_key = 'group_compute_ms'
+
+                project_rec[count_key] += entry['count']
+                project_rec[time_key] += entry['total_ms']
+
+        # Bulk updates
```
Comment (hkethi002) [inline api/reports/usage_report.py]: I would break this out into its own function as well

**Comment by ehlertjd** [comment]
@hkethi002 Created [FLYW-1718](https://flywheelio.atlassian.net/browse/FLYW-1718) to track the minor refactor work.

**Pull request merged**